avm/res/service-fabric/cluster/main.bicep

```diff
@@ -134,98 +134,163 @@ param applicationTypes array = []
+          startPort: contains(nodeType.applicationPorts, 'startPort') ? nodeType.applicationPorts.startPort : null
+        }
+      : null
+    capacities: contains(nodeType, 'capacities') ? nodeType.capacities : null
+    clientConnectionEndpointPort: contains(nodeType, 'clientConnectionEndpointPort')
+      ? nodeType.clientConnectionEndpointPort
+      : null
+    durabilityLevel: contains(nodeType, 'durabilityLevel') ? nodeType.durabilityLevel : null
+    ephemeralPorts: contains(nodeType, 'ephemeralPorts')
+      ? {
+          endPort: contains(nodeType.ephemeralPorts, 'endPort') ? nodeType.ephemeralPorts.endPort : null
+          startPort: contains(nodeType.ephemeralPorts, 'startPort') ? nodeType.ephemeralPorts.startPort : null
+        }
+      : null
+    httpGatewayEndpointPort: contains(nodeType, 'httpGatewayEndpointPort') ? nodeType.httpGatewayEndpointPort : null
+    isPrimary: contains(nodeType, 'isPrimary') ? nodeType.isPrimary : null
+    isStateless: contains(nodeType, 'isStateless') ? nodeType.isStateless : null
+    multipleAvailabilityZones: contains(nodeType, 'multipleAvailabilityZones')
+      ? nodeType.multipleAvailabilityZones
+      : null
+    name: contains(nodeType, 'name') ? nodeType.name : 'Node00'
+    placementProperties: contains(nodeType, 'placementProperties') ? nodeType.placementProperties : null
+    reverseProxyEndpointPort: contains(nodeType, 'reverseProxyEndpointPort') ? nodeType.reverseProxyEndpointPort : null
+    vmInstanceCount: contains(nodeType, 'vmInstanceCount') ? nodeType.vmInstanceCount : 1
+  }
+]
+
+var notificationsVar = [
+  for notification in notifications: {
+    isEnabled: contains(notification, 'isEnabled') ? notification.isEnabled : false
+    notificationCategory: contains(notification, 'notificationCategory')
+      ? notification.notificationCategory
+      : 'WaveProgress'
+    notificationLevel: contains(notification, 'notificationLevel') ? notification.notificationLevel : 'All'
+    notificationTargets: contains(notification, 'notificationTargets') ? notification.notificationTargets : []
+  }
+]
+
+var upgradeDescriptionVar = union(
+  {
     deltaHealthPolicy: {
-      applicationDeltaHealthPolicies: contains(upgradeDescription, 'applicationDeltaHealthPolicies') ? upgradeDescription.applicationDeltaHealthPolicies : {}
-      maxPercentDeltaUnhealthyApplications: contains(upgradeDescription, 'maxPercentDeltaUnhealthyApplications') ? upgradeDescription.maxPercentDeltaUnhealthyApplications : 0
-      maxPercentDeltaUnhealthyNodes: contains(upgradeDescription, 'maxPercentDeltaUnhealthyNodes') ? upgradeDescription.maxPercentDeltaUnhealthyNodes : 0
-      maxPercentUpgradeDomainDeltaUnhealthyNodes: contains(upgradeDescription, 'maxPercentUpgradeDomainDeltaUnhealthyNodes') ? upgradeDescription.maxPercentUpgradeDomainDeltaUnhealthyNodes : 0
+      applicationDeltaHealthPolicies: contains(upgradeDescription, 'applicationDeltaHealthPolicies')
```

Comment:
It's a minor thing / suggestion so feel free to just resolve it, but if you want to simply these statements you can use the tryGet assert (`?`) in combination with the coalesce (`??`)

For example, instead of 
```bicep
maxPercentDeltaUnhealthyNodes: contains(upgradeDescription, 'maxPercentDeltaUnhealthyNodes')
        ? upgradeDescription.maxPercentDeltaUnhealthyNodes
        : 0
```

you could simply it to
```bicep
maxPercentDeltaUnhealthyNodes: upgradeDescription.?maxPercentDeltaUnhealthyNodes ?? 0
```

Reply:
Again, no need to 'actually do it'. Just figured I should mention it as the formatter took a swing at it 😄